src/twisted/internet/interfaces.py

```diff
@@ -1807,11 +1807,12 @@ def dataReceived(data):
         callback will be made upon the receipt of each complete protocol
         message.
 
-        @param data: a string of indeterminate length.  Please keep in mind
-            that you will probably need to buffer some data, as partial
-            (or multiple) protocol messages may be received!  I recommend
-            that unit tests for protocols call through to this method with
-            differing chunk sizes, down to one byte at a time.
+        Please keep in mind that you will probably need to buffer some data
+        as partial (or multiple) protocol messages may be received!  I
+        recommend that unit tests for protocols call through to this method
+        with differing chunk sizes, down to one byte at a time.
```

Comment:
Because it is not a string, it is bytes.  The fact that it is bytes is covered by the @type field.  Once you have that, then the only thing to put in the @param field would be something like
**@param data: the data that is received** which is redundant.  The rest of the long description can go in the docstring outside of any @param tag